### DISCUSSION TITLE
Small angle approximation effect on the results

### DISCUSSION THREAD

**Discussion by MYMahfouz** [OpenFAST/openfast#2228] [Q&A] [answered]
Dear all,

I am modeling a FOWT with a soft mooring system and low yaw stiffness. I am aware that increasing the yaw stiffness of the MS will solve the following issue, but my work includes checking the effect of the mooring stiffness on the FOWT loads.

Due to the more significant yaw aerodynamic moment, the low mooring system stiffness allows larger yaw angles, especially at higher wind speeds of 24-25 m/s.

For these cases, I am getting the warning, but the simulation does not fail or abort.

> FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption1:ED_HD_InputOutputSolve:HydroDyn_CalcOutp
ut: Angles in GetSmllRotAngs() are larger than 0.4 radians.
HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are larger than 0.4 radians.
ED_HD_InputOutputSolve:HydroDyn_CalcOutput: Angles in GetSmllRotAngs() are larger than 0.4
radians.
HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are larger than 0.4 radians.

I know that OpenFAST uses small angle approximation for the rotational transformation matrix. 

I have also seen in some references with OpenFAST to vortex wake model comparisons and LES that OpenFAST overestimates the aerodynamic yaw moment at high values of yaw misalignment. However, these comparisons are for fixed bottom turbines.

In my simulations, I also see couplings between the roll, pitch, and yaw DoFs for the platform motions as the platform yaw angle exceeds 15 degrees. This leads to higher loads on the tower base due to the larger amplitude of the roll and pitch oscillations.

My questions are: 
- What is the effect of the small angle approximation for a FOWT? Does it lead to overestimating the yaw moment, similar to the fixed bottom?
- Are the roll, pitch, and yaw couplings at high yaw angle values due to the small angle approximation? Is it because the decomposition of the mean rotor thrust in the fixed reference frame x and y directions also assumes a small angle approximation?
- Would you happen to have any recommendations to help me analyze the system better?

Best regards,

**Comment by jjonkman** [comment]
Dear @MYMahfouz,

FYI: OpenFAST currently uses **small** angle approximations for platform roll-pitch-yaw rotations, with a correction for orthogonality that enables support for moderate angles.

I'm not sure that I can say whether the error resulting from large platform-yaw angles would result in an under- or overestimation of the yaw moment; this may depend on the conditions.  Regardless, I would not trust the solution accuracy for the cases that trigger this warning.

I'm also not sure I understand your second question, but again, the results are likely not accurate/reliable if the platform-yaw angle gets large.

I'm not sure what condition you are trying to simulate, but is there a way to limit the yaw motion within a given simulation?

FYI: We are currently working on an enhancement to OpenFAST to support large platform-yaw motion, which is important for modeling single-point mooring systems and load cases involving mooring line loss, both of which may have large platform-yaw motion.  We just submitted an abstract to present such results at the upcoming NAWEA / WindTech 2024 conference.  You can track the code-development progress in the following OpenFAST PR: https://github.com/OpenFAST/openfast/pull/2203.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,

I appreciate your detailed reply.

Sadly, there is no other way to check the effect of the low MS stiffness on the system loads.

I am simulating operating conditions with a mean wind speed of 25 m/s and turbulence intensity of 6%.

In my second question, I was referring to a coupling between the roll, pitch, and yaw motions that occurs if the yaw angle gets larger and the error about the small angle approximation is printed out.
I believe the error occurs due to aerodynamic coupling between roll, pitch, and yaw DoF. I am not sure if it has anything to do with the small angle approximation within OpenFAST or not. It does not make the simulation unstable; it just significantly increases the tower base load due to the higher roll pitch oscillation.
Turning off either the roll or the pitch DoF makes the coupling disappear, and no high amplitude oscillation occurs. Also, increasing either the roll or pitch damping in HydroDyn leads to the same effect where the coupling is no longer visible.

In my current case, the yaw aerodynamic moment increases significantly if the yaw angle becomes large.


I look forward to your comping paper with the results and OpenFASt v4.0.

Best regards,

**Reply by jjonkman** [reply, marked as answer]
Dear @MYMahfouz,

I can certainly foresee that the roll and pitch motions will be inaccurate if the yaw angle is large because of the small/moderate angle assumptions.

Actually, we expect OpenFAST v4.0 to be released soon, but the support for large platform yaw will not be released until v5.0 or v5.1 later this fall.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,

Thank you. This was also my understanding.

I will recheck these results whenever the large platform yaw is supported in OpenFAST.


Best regards,
